test/common/testlib.py

```diff
@@ -2092,6 +2094,34 @@ def enable_root_login(self):
         if not self.machine.ostree_image and self.file_exists(disallowed_conf):
             self.sed_file('/root/d', disallowed_conf)
 
+###########################
+
+
+
+class ExtendedMachineCase(unittest.TestCase):
+    def reboot(self, timeout_sec=None):
+        self.allow_restart_journal_messages()
+        self.machine.reboot()
+        if timeout_sec is not None:
+            self.machine.wait_for_ssh(timeout=timedelta(seconds=timeout_sec))
+
+    def wait_reboot(self, timeout_sec=None):
+        self.allow_restart_journal_messages()
+        self.machine.wait_reboot()
+        if timeout_sec is not None:
+            self.machine.wait_for_ssh(timeout=timedelta(seconds=timeout_sec))
+
+
+
+class MyTestCase(ExtendedMachineCase):
+    def test_reboot_with_timeout(self):
+        self.reboot(timeout_sec=30)
+
+    def test_wait_reboot_with_timeout(self):
+        self.wait_reboot(timeout_sec=30)
```

Comment:
If this worked, the test would reboot the test machine, that's not really what you want I think. It also does not really `assert` anything. So in essence this testcase is not testing anything, if you want to really test if the function rebooted, you could compare for example the `journalctl --list-boots` before and after.

Reply:
Please drop the whole MyTestCase. Self-tests don't belong into testlib.py (there's a separate place for that, test/verify/check-testlib.py), and this also does not need an explicit test case. Rather just convert the existing test cases to use the new API, and if they pass, that'll be enough proof. E.g. change test/verify/check-system-shutdown-restart to use the new API.